src/rpc/misc.cpp

```diff
@@ -366,6 +368,34 @@ static UniValue setmocktime(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static UniValue mockscheduler(const JSONRPCRequest& request)
+{
+        RPCHelpMan{"mockscheduler",
+            "\nBump the scheduler into the future (-regtest only)\n",
+            {
+                {"delta_time", RPCArg::Type::NUM, RPCArg::Optional::NO, "Number of seconds to forward the scheduler into the future." },
+            },
+            RPCResults{},
+            RPCExamples{""},
+        }.Check(request);
+
+    if (!Params().MineBlocksOnDemand()) {
```

Comment:
I realize this is copied from `setmocktime`, but it feels very strange to use MineBlocksOnDemand to gate time mocking. Perhaps it's worth adding a separate chainparams property for time mocking, and then using it for both `setmocktime` and `mockscheduler`.

Reply:
@sipa what would you think about just abstracting out a new function named `AllowMockableTime()` and having it call MineBlocksOnDemand  -- that way someone can clean up the logic later, I'm sure we're doing other weird things like this that could also be fixed.

Reply:
Or just making it the responsibility of follow up work, I don't think it's that bad (if someone reviewed using MineBlocksOnDemand for this once... it should still be fine now)

Reply:
I agree that using `MineBlocksOnDemand` felt weird. Added `m_is_mockable_chain` to chainparams & implemented for `mockscheduler`. If it looks okay I'll do a follow up PR to use that for `mocktime`.